src/dropbox.js

```diff
@@ -16,9 +23,27 @@ const dbx = {
         } else  {
             config.accessToken = accessToken;
             this.app = new dropbox.Dropbox(config);
+            if (!this.isDropboxReady() && !scriptLoadingStarted) {
+                scriptLoadingStarted = true;
+                loadScript(DROPBOX_SDK_URL, {
+                  attrs : {
+                    id: SCRIPT_ID,
+                    'data-app-key': config.appKey
+                  }
+                });
+              }
         }
     },
 
+    isDropboxReady() {
+        return !!window.Dropbox;
+    },
+
+    onChoose(options) {
```

Comment:
Have onChoose take in file type and callback and resolve options in here

Reply:
done